packages/chain-adapters/src/bitcoin/BitcoinChainAdapter.ts

```diff
@@ -271,30 +269,50 @@ export class ChainAdapter implements IChainAdapter<ChainTypes.Bitcoin> {
       throw new Error('undefined fee')
 
     // We have to round because coinselect library uses sats per byte which cant be decimals
-    const fast = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
-    const average = String(Math.round(feeData.data.average?.satsPerKiloByte / 1024))
-    const slow = String(Math.round(feeData.data.slow?.satsPerKiloByte / 1024))
+    const fastPerByte = String(Math.round(feeData.data.fast?.satsPerKiloByte / 1024))
+    const averagePerByte = String(Math.round(feeData.data.average?.satsPerKiloByte / 1024))
+    const slowPerByte = String(Math.round(feeData.data.slow?.satsPerKiloByte / 1024))
+
+    const { data: utxos } = await this.providers.http.getUtxos({
+      pubkey
+    })
+
+    type MappedUtxos = Omit<bitcoin.api.Utxo, 'value'> & { value: number }
+    const mappedUtxos: MappedUtxos[] = utxos.map((x) => ({ ...x, value: Number(x.value) }))
+
+    const { fee: fastFee } = coinSelect<MappedUtxos, chainAdapters.bitcoin.Recipient>(
+      mappedUtxos,
+      [{ value: Number(value), address: to }],
+      Number(fastPerByte)
+    )
+    const { fee: averageFee } = coinSelect<MappedUtxos, chainAdapters.bitcoin.Recipient>(
+      mappedUtxos,
+      [{ value: Number(value), address: to }],
+      Number(averagePerByte)
+    )
+    const { fee: slowFee } = coinSelect<MappedUtxos, chainAdapters.bitcoin.Recipient>(
+      mappedUtxos,
+      [{ value: Number(value), address: to }],
+      Number(slowPerByte)
+    )
```

Comment:
I would recommend using the helper function "bnOrZero" which will return 0 if the input is not valid and returns a BigNumber.

This will guard against unexpected values and make sure we don't get weird math that ends up as NaN.

Reply:
Are you sure its needed here? Your saying fastFee/averageFee/slowFee should bet set in a bnOrZero?

In some cases I feel like setting to 0 instead of nan could be unintentionally hiding errors